### PR TITLE
Add description for "Two neuron example"

### LINKED ISSUE #1754: Add description for "Two neuron example"
The ``twoneurons.py`` example is currently lacking a description and instructions (please see ``one_neuron.py`` as example).

One neuron: https://github.com/nest/nest-simulator/blob/master/pynest/examples/one_neuron.py
Two neuron: https://github.com/nest/nest-simulator/blob/master/pynest/examples/twoneurons.py

### PR DESCRIPTION
**Pull request by YounesBouhadjar**
fixes #1754
and minor corrections of `one_neuron.py` documentation

### CODE REVIEW

File: pynest/examples/twoneurons.py [hunk 1]
```diff
@@ -22,32 +22,62 @@
 """Two neuron example
 ----------------------------
 
+This script simulates two connected pre and postsynaptic neurons.
```
Comment (sarakonradi) [inline pynest/examples/twoneurons.py:25]:
```suggestion
This script simulates two connected pre and postsynaptic neurons.
```

File: pynest/examples/twoneurons.py [hunk 2]
```diff
@@ -22,32 +22,62 @@
...
+neuron_2 = nest.Create("iaf_psc_alpha")
+voltmeter = nest.Create("voltmeter")
+
+###############################################################################
+# Third, we set the external current of neuron 1.
+
+neuron_1.I_e = 376.0
+
+###############################################################################
+# Fourth, we connect neuron 1 to neuron 2.
```
Comment (sarakonradi) [inline pynest/examples/twoneurons.py:59]:
```suggestion
# Fourth, we connect neuron 1 to neuron 2.
```

File: pynest/examples/twoneurons.py [hunk 3]
```diff
@@ -22,32 +22,62 @@
...
-nest.Simulate(100.0)
+nest.Simulate(1000.0)
+
+###############################################################################
+# Finally, we plot the neurons's membrane potential as a function of
+# time.
 
 nest.voltage_trace.from_device(voltmeter)
 plt.show()
+
```
Comment (sarakonradi) [inline pynest/examples/twoneurons.py:83]:
```suggestion
plt.show()
```

File: pynest/examples/twoneurons.py (hunk 1 above)
Comment (sarakonradi) [inline pynest/examples/twoneurons.py:25]:
```suggestion
This script simulates two connected pre- and postsynaptic neurons. 
```

File: pynest/examples/twoneurons.py [hunk 4]
```diff
@@ -22,32 +22,62 @@
...
+voltmeter = nest.Create("voltmeter")
+
+###############################################################################
+# Third, we set the external current of neuron 1.
+
+neuron_1.I_e = 376.0
+
+###############################################################################
+# Fourth, we connect neuron 1 to neuron 2. 
+# And, then we connect a voltmeter to the two neurons.
```
Comment (sarakonradi) [inline pynest/examples/twoneurons.py:60]:
```suggestion
# Then, we connect a voltmeter to the two neurons.
```

File: pynest/examples/twoneurons.py [hunk 5]
```diff
@@ -22,32 +22,62 @@
...
-nest.Connect(voltmeter, neuron2)
+###############################################################################
+# Now we simulate the network using ``Simulate``, which takes the
+# desired simulation time in milliseconds.
 
-nest.Simulate(100.0)
+nest.Simulate(1000.0)
+
+###############################################################################
+# Finally, we plot the neurons's membrane potential as a function of
```
Comment (sarakonradi) [inline pynest/examples/twoneurons.py:78]:
```suggestion
# Finally, we plot the neurons' membrane potential as a function of
```

**Review by sarakonradi — CHANGES_REQUESTED** [review]
@YounesBouhadjar Thanks a lot for this! Looks good overall. Just added some minor suggestions and PEP8-related comments.

File: pynest/examples/twoneurons.py [hunk 6]
```diff
@@ -22,32 +22,61 @@
 """Two neuron example
 ----------------------------
 
+This script simulates two connected pre and postsynaptic neurons.
```
Comment (sarakonradi) [inline pynest/examples/twoneurons.py:25]:
```suggestion
This script simulates two connected pre- and postsynaptic neurons.
```

**Review by sarakonradi — APPROVED** [review]
LGTM :+1: 

It is passing the Travis checks, apart from the usual macOS issues. @hakonsbm What do you think?

File: pynest/examples/twoneurons.py [hunk 7]
```diff
@@ -22,32 +22,61 @@
 """Two neuron example
 ----------------------------
 
+This script simulates two connected pre- and postsynaptic neurons.
+The presynaptic neuron receives a constant external current.
+And, the membrane potential of both neurons are recorded.
```
Comment (hakonsbm) [inline pynest/examples/twoneurons.py:27]: Isn't starting a sentence with `And,` bad practice? I suggest joining it with the previous sentence.
```suggestion
This script simulates two connected pre- and postsynaptic neurons.
The presynaptic neuron receives a constant external current,
and the membrane potential of both neurons are recorded.
```

File: pynest/examples/twoneurons.py [hunk 8]
```diff
@@ -22,32 +22,61 @@
...
 
+###############################################################################
+# First, we import all necessary modules for simulation, analysis and plotting.
 
 import nest
 import nest.voltage_trace
 import matplotlib.pyplot as plt
 
+nest.set_verbosity("M_WARNING")
+nest.ResetKernel()
```
Comment (hakonsbm) [inline pynest/examples/twoneurons.py:44]: Should probably add something about these to not confuse new users.

File: pynest/examples/twoneurons.py [hunk 9]
```diff
@@ -22,32 +22,61 @@
...
+# :doc:`one neuron example <one_neuron>`.
+
 weight = 20.0
 delay = 1.0
-stim = 1000.0
 
-neuron1 = nest.Create("iaf_psc_alpha")
-neuron2 = nest.Create("iaf_psc_alpha")
-voltmeter = nest.Create("voltmeter")
+nest.Connect(neuron_1, neuron_2, syn_spec={'weight': weight, 'delay': delay})
```
Comment (hakonsbm) [inline pynest/examples/twoneurons.py:67]: While it's allowed to use both single and double quotes for strings, it should be consistent. We are using double in the single neuron example, so let's go with double here too.
```suggestion
nest.Connect(neuron_1, neuron_2, syn_spec={"weight": weight, "delay": delay})
```

**Review by hakonsbm — CHANGES_REQUESTED** [review]
Thanks! I have just a couple small suggestions.

File: pynest/examples/twoneurons.py (hunk 8 above)
Comment (YounesBouhadjar) [reply pynest/examples/twoneurons.py:46]: thanks, just added a description about that

File: pynest/examples/twoneurons.py [hunk 10]
```diff
@@ -22,32 +22,63 @@
...
-~~~~~~~~~~
+~~~~~~~~
 
 :doc:`one_neuron`
 
 """
 
+###############################################################################
+# First, we import all necessary modules for simulation, analysis and plotting.
+# Additionally, we set the verbosity to suppress info messages and reset
```
Comment (hakonsbm) [inline pynest/examples/twoneurons.py:38]:
```suggestion
# Additionally, we set the verbosity to suppress info messages and reset
```

**Review by hakonsbm — CHANGES_REQUESTED** [review]

**Review by hakonsbm — APPROVED** [review]
Looks good now, thanks!

**Comment by hakonsbm** [comment]
This only affects Python documentation, so we can ignore the failing macOS CI stage (see #1822) and merge.

**Pull request merged**